<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>539</width>
    <height>550</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>DFRobot - I2C Motors Remote Control</string>
  </property>
  <property name="statusTip">
   <string/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="arduinoConnectionGroupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>211</width>
      <height>321</height>
     </rect>
    </property>
    <property name="title">
     <string>Arduino connection</string>
    </property>
    <widget class="QPushButton" name="pushButton_updateList">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>191</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Update list</string>
     </property>
    </widget>
    <widget class="QComboBox" name="comboBox_serialSlots">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>191</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_connectArduino">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>191</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Connect</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_disconnectArduino">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>191</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Disconnect</string>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="arduinoConnectionConsole">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>191</width>
       <height>131</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Bitstream Charter</family>
       <pointsize>8</pointsize>
       <weight>50</weight>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
       <kerning>false</kerning>
      </font>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="lineWidth">
      <number>1</number>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="tabChangesFocus">
      <bool>false</bool>
     </property>
     <property name="plainText">
      <string/>
     </property>
     <property name="textInteractionFlags">
      <set>Qt::NoTextInteraction</set>
     </property>
     <property name="backgroundVisible">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="statusLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>290</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Status:</string>
     </property>
    </widget>
    <widget class="QLabel" name="arduinoConnectionStatusLabel">
     <property name="geometry">
      <rect>
       <x>70</x>
       <y>290</y>
       <width>121</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>NOT CONNECTED</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="rcConnectionGroupBox">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>0</y>
      <width>301</width>
      <height>321</height>
     </rect>
    </property>
    <property name="sizePolicy">
     <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
      <horstretch>0</horstretch>
      <verstretch>0</verstretch>
     </sizepolicy>
    </property>
    <property name="title">
     <string>RC Connection</string>
    </property>
    <widget class="QPushButton" name="startServerButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>131</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Start server</string>
     </property>
    </widget>
    <widget class="QLabel" name="rcConnectionServerLabel">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>290</y>
       <width>51</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Server:</string>
     </property>
    </widget>
    <widget class="QLabel" name="rcConnectionServerStatus">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>290</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>STOPPED</string>
     </property>
    </widget>
    <widget class="QPushButton" name="stopServerButton">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>30</y>
       <width>131</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Stop server</string>
     </property>
    </widget>
    <widget class="QPlainTextEdit" name="rcConnectionConsoleTextBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>150</y>
       <width>281</width>
       <height>131</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Bitstream Charter</family>
       <pointsize>8</pointsize>
       <weight>50</weight>
       <bold>false</bold>
       <underline>false</underline>
       <strikeout>false</strikeout>
       <kerning>false</kerning>
      </font>
     </property>
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Sunken</enum>
     </property>
     <property name="lineWidth">
      <number>1</number>
     </property>
     <property name="tabChangesFocus">
      <bool>false</bool>
     </property>
     <property name="plainText">
      <string/>
     </property>
     <property name="textInteractionFlags">
      <set>Qt::NoTextInteraction</set>
     </property>
     <property name="backgroundVisible">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="rcConnectionIPAddressTextBox">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>131</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>localhost</string>
     </property>
     <property name="readOnly">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QLineEdit" name="rcConnectionPortTextBox">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>60</y>
       <width>131</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
      </font>
     </property>
     <property name="text">
      <string>50500</string>
     </property>
     <property name="readOnly">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="rcConnectionclientLabel">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>290</y>
       <width>41</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Client:</string>
     </property>
    </widget>
    <widget class="QLabel" name="rcConnectionClientStatus">
     <property name="geometry">
      <rect>
       <x>174</x>
       <y>290</y>
       <width>115</width>
       <height>21</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <pointsize>10</pointsize>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>NOT CONNECTED</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="motorsControlGroupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>320</y>
      <width>521</width>
      <height>171</height>
     </rect>
    </property>
    <property name="title">
     <string>Motors Control</string>
    </property>
    <widget class="QLabel" name="motor1_speedLabel">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>30</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="motor4_speedLabel">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>120</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="m4Label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>120</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>M4</string>
     </property>
    </widget>
    <widget class="QLabel" name="m2Label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>M2</string>
     </property>
    </widget>
    <widget class="QLabel" name="motor2_speedLabel">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>60</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QLabel" name="motor3_speedLabel">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>90</y>
       <width>31</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>0</string>
     </property>
    </widget>
    <widget class="QSlider" name="motor3_slider">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>90</y>
       <width>351</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_reverse3">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>90</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Reverse</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QSlider" name="motor4_slider">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>120</y>
       <width>351</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_reverse4">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>120</y>
       <width>71</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Reverse</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QSlider" name="motor1_slider">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>30</y>
       <width>351</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QLabel" name="m3Label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>90</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>M3</string>
     </property>
    </widget>
    <widget class="QLabel" name="m1Label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>21</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>M1</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_reverse1">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>30</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Reverse</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QSlider" name="motor2_slider">
     <property name="geometry">
      <rect>
       <x>40</x>
       <y>60</y>
       <width>351</width>
       <height>21</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_reverse2">
     <property name="geometry">
      <rect>
       <x>440</x>
       <y>60</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string>Reverse</string>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_minMotors">
     <property name="geometry">
      <rect>
       <x>140</x>
       <y>140</y>
       <width>89</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Min</string>
     </property>
    </widget>
    <widget class="QPushButton" name="pushButton_maxMotors">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>140</y>
       <width>89</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Max</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>539</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_connectArduino</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>pushButton_connect_onClick()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>210</x>
     <y>129</y>
    </hint>
    <hint type="destinationlabel">
     <x>437</x>
     <y>118</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_connectArduino</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_pushButton_connectArduino_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>139</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>227</x>
     <y>173</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_maxMotors</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>on_pushButton_maxMotors_clicked()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>283</x>
     <y>511</y>
    </hint>
    <hint type="destinationlabel">
     <x>561</x>
     <y>525</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>on_pushButton_maxMotors_clicked()</slot>
 </slots>
</ui>
